src/gui/widgets/FxLine.cpp

```diff
@@ -94,13 +93,37 @@ FxLine::FxLine( QWidget * _parent, FxMixerView * _mv, int _channelIndex) :
 	"to the channel.\n\n"
 	
 	"You can remove and move FX channels in the context menu, which is accessed "
-	"by right-clicking the FX channel.\n") );
+	"by right-clicking the FX channel.\n" ) );
+
+	m_mix = Engine::fxMixer();
+	m_name = m_mix->effectChannel( m_channelIndex )->m_name;
+	setToolTip( m_name );
+
+	m_renameLineEdit = new QLineEdit();
+	m_renameLineEdit->setText( m_name );
+	m_renameLineEdit->setFixedWidth( 65 );
+	m_renameLineEdit->setFont( pointSizeF( FxLine::font(), 7.5f ) );
+	m_renameLineEdit->setReadOnly( true );
 
-	FxMixer * mix = Engine::fxMixer();
-	setToolTip( mix->effectChannel( m_channelIndex )->m_name );
+	m_scene = new QGraphicsScene();
+	m_scene->setSceneRect( 0, 0, 33, FxLineHeight );
+
+	m_view = new QGraphicsView( this );
+	m_view->setStyleSheet( "border-style: none; background: transparent;" );
```

Comment:
Just to let you know: if a track or a fx is renamed Lmms does not consider the project as modified.
You can quit with no question about saving...